### ISSUE TITLE
 Changing Per Page dropdown does not change the actual number of items in the listview

### ISSUE THREAD

**Issue by quazardous** [sonata-project/SonataAdminBundle#5484]
### Environment
Fedora 29

#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
# Put the result here.

sonata-project/admin-bundle              3.45.2 3.45.2 The missing Symfony Admin Generator
sonata-project/block-bundle              3.14.0 3.14.0 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.16.1 3.16.1 Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.4.0  2.4.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.1.5  1.1.5  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.8.2  3.8.2  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.5.0  2.5.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  2.0.1  2.0.1  Lightweight Exporter library
sonata-project/intl-bundle               2.5.0  2.5.0  Symfony SonataIntlBundle
sonata-project/user-bundle               4.3.0  4.3.0  Symfony SonataUserBundle

```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
# Put the result here.

symfony/asset                 v4.2.3  v4.2.3  Symfony Asset Component
symfony/browser-kit           v4.2.3  v4.2.3  Symfony BrowserKit Component
symfony/cache                 v4.2.3  v4.2.3  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/config                v4.2.3  v4.2.3  Symfony Config Component
symfony/console               v4.2.3  v4.2.3  Symfony Console Component
symfony/contracts             v1.0.2  v1.0.2  A set of abstractions extracted out of the Symfony components
symfony/css-selector          v4.2.3  v4.2.3  Symfony CssSelector Component
symfony/debug                 v4.2.3  v4.2.3  Symfony Debug Component
symfony/debug-bundle          v4.2.3  v4.2.3  Symfony DebugBundle
symfony/dependency-injection  v4.2.3  v4.2.3  Symfony DependencyInjection Component
symfony/doctrine-bridge       v4.2.3  v4.2.3  Symfony Doctrine Bridge
symfony/dom-crawler           v4.2.3  v4.2.3  Symfony DomCrawler Component
symfony/dotenv                v4.2.3  v4.2.3  Registers environment variables from a .env file
symfony/event-dispatcher      v4.2.3  v4.2.3  Symfony EventDispatcher Component
symfony/expression-language   v4.2.3  v4.2.3  Symfony ExpressionLanguage Component
symfony/filesystem            v4.2.3  v4.2.3  Symfony Filesystem Component
symfony/finder                v4.2.3  v4.2.3  Symfony Finder Component
symfony/flex                  v1.2.0  v1.2.0  Composer plugin for Symfony
symfony/form                  v4.2.3  v4.2.3  Symfony Form Component
symfony/framework-bundle      v4.2.3  v4.2.3  Symfony FrameworkBundle
symfony/http-foundation       v4.2.3  v4.2.3  Symfony HttpFoundation Component
symfony/http-kernel           v4.2.3  v4.2.3  Symfony HttpKernel Component
symfony/inflector             v4.2.3  v4.2.3  Symfony Inflector Component
symfony/intl                  v4.2.3  v4.2.3  A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/maker-bundle          v1.11.3 v1.11.3 Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about writ...
symfony/monolog-bridge        v4.2.3  v4.2.3  Symfony Monolog Bridge
symfony/monolog-bundle        v3.3.1  v3.3.1  Symfony MonologBundle
symfony/options-resolver      v4.2.3  v4.2.3  Symfony OptionsResolver Component
symfony/orm-pack              v1.0.6  v1.0.6  A pack for the Doctrine ORM
symfony/panther               v0.3.0  v0.3.0  A browser testing and web scraping library for PHP and Symfony.
symfony/phpunit-bridge        v4.2.3  v4.2.3  Symfony PHPUnit Bridge
symfony/polyfill-apcu         v1.10.0 v1.10.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu     v1.10.0 v1.10.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring     v1.10.0 v1.10.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72        v1.10.0 v1.10.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/process               v4.2.3  v4.2.3  Symfony Process Component
symfony/profiler-pack         v1.0.4  v1.0.4  A pack for the Symfony web profiler
symfony/property-access       v4.2.3  v4.2.3  Symfony PropertyAccess Component
symfony/property-info         v4.2.3  v4.2.3  Symfony Property Info Component
symfony/routing               v4.2.3  v4.2.3  Symfony Routing Component
symfony/security-acl          v3.0.1  v3.0.1  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle       v4.2.3  v4.2.3  Symfony SecurityBundle
symfony/security-core         v4.2.3  v4.2.3  Symfony Security Component - Core Library
symfony/security-csrf         v4.2.3  v4.2.3  Symfony Security Component - CSRF Library
symfony/security-guard        v4.2.3  v4.2.3  Symfony Security Component - Guard
symfony/security-http         v4.2.3  v4.2.3  Symfony Security Component - HTTP Integration
symfony/serializer            v4.2.3  v4.2.3  Symfony Serializer Component
symfony/stopwatch             v4.2.3  v4.2.3  Symfony Stopwatch Component
symfony/swiftmailer-bundle    v3.2.5  v3.2.5  Symfony SwiftmailerBundle
symfony/templating            v4.2.3  v4.2.3  Symfony Templating Component
symfony/test-pack             v1.0.5  v1.0.5  A pack for functional and end-to-end testing within a Symfony app
symfony/translation           v4.2.3  v4.2.3  Symfony Translation Component
symfony/twig-bridge           v4.2.3  v4.2.3  Symfony Twig Bridge
symfony/twig-bundle           v4.2.3  v4.2.3  Symfony TwigBundle
symfony/validator             v4.2.3  v4.2.3  Symfony Validator Component
symfony/var-dumper            v4.2.3  v4.2.3  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter          v4.2.3  v4.2.3  A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-profiler-bundle   v4.2.3  v4.2.3  Symfony WebProfilerBundle
symfony/webpack-encore-bundle v1.1.0  v1.1.0  Integration with your Symfony app & Webpack Encore!
symfony/yaml                  v4.2.3  v4.2.3  Symfony Yaml Component

```

#### PHP version

```
$ php -v
# Put the result here.

PHP 7.3.2 (cli) (built: Feb  5 2019 13:10:03) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.2, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.2, Copyright (c) 1999-2018, by Zend Technologies

```

## Subject

Changing **Per Page** dropdown does not change the actual number of items in the listview.

## Steps to reproduce

Go on a big List. Select something else than 32.

## Expected results

The list should display the required number of items.

## Actual results

It's stuck to 32 (the dropdown does not keep the chosen value AND the number of items is wrong)

but the URL has the good `_per_page` value.

```
.../admin/app/game/list?filter[_sort_order]=DESC&filter[_page]=1&filter[_per_page]=256&...
```

**Comment by axzx** [comment]
I confirm. 
I fixed it here
https://github.com/sonata-project/SonataAdminBundle/pull/5481

**Comment by Krleza** [comment]
Same Here. Fix sonata-project/admin-bundle to version 3.45.1 solves the issue. 
version 3.45.2 and above have the bug.

**Comment by mkmieciakPl** [comment]
The strict option in determinedPerPageValue method in AbstractAdmin should be set to false.
